src/main/java/uk/gov/digital/ho/hocs/queue/migration/MigrationValidator.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.digital.ho.hocs.queue.migration;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.queue.complaints.ukvi.UKVITypeData;
+
+import java.io.InputStream;
+import java.util.Set;
+
+@Slf4j
+@Service
+public class MigrationValidator {
+
+    private final ObjectMapper objectMapper;
+    private final JsonSchema schema;
+
+    @Autowired
+    public MigrationValidator(ObjectMapper objectMapper) {
+        InputStream in = getClass().getResourceAsStream("/hocs-migration-schema.json");
```

Comment:
This is the second validator within the system, I'm starting to think this JSON Reader should be pulled into a abstract class.